src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -1301,9 +1303,16 @@ impl CgroupManager for Manager {
         };
 
         // BlkioStats
+        // note that virtiofs has no blkio stats
         info!(sl!(), "blkio_stats");
         let blkio_stats = if self.paths.get("blkio").is_some() {
-            SingularPtrField::some(Blkio().get_stats(self.paths.get("blkio").unwrap())?)
+            match Blkio().get_stats(self.paths.get("blkio").unwrap()) {
+                Ok(stat) => SingularPtrField::some(stat),
+                Err(e) => {
+                    warn!(sl!(), "failed to get blkio stats");
+                    SingularPtrField::none()
+                }
```

Comment:
Are you sure to return none when it errors?

Reply:
Yes, since maybe some of the stats didn't supported, which would cause the getstats failed,and thus we'd better return none instead of return error.